src/test/scala/fundamentals/level03/ExceptionExercisesTest.scala

```diff
@@ -106,15 +106,23 @@ class ExceptionExercisesTest extends FunSpec with TypeCheckedTripleEquals {
 
   describe("collectErrors") {
 
-    it("should return a List Exceptions thrown while processing inputs") {
-
-      def exceptionEq(e1: Exception, e2: Exception): Boolean =
-        e1.getClass == e2.getClass && e1.getMessage == e2.getMessage
+    def exceptionEq(e1: Exception, e2: Exception): Boolean =
+      e1.getClass == e2.getClass && e1.getMessage == e2.getMessage
+
+    val expectedErrors = List(
+      new InvalidAgeValueException("provided age is invalid: 5o"),
+      new InvalidAgeRangeException("provided age should be between 1-120: 200"),
+      new InvalidAgeRangeException("provided age should be between 1-120: 0"),
+      new EmptyNameException("provided name is empty"),
+      new InvalidAgeRangeException("provided age should be between 1-120: 1000"),
+      new EmptyNameException("provided name is empty")
+    )
+
+    it("should return all errors") {
+      collectErrors.size === 6
+    }
 
-      val expectedErrors = List(new InvalidAgeValueException("provided age is invalid: 5o"),
-                                    new InvalidAgeRangeException("provided age should be between 1-120: 200"),
-                                    new InvalidAgeRangeException("provided age should be between 1-120: 0"),
-                                    new EmptyNameException("provided name is empty"))
+    it("should return a List Exceptions thrown while processing inputs") {
       collectErrors.zip(expectedErrors).foreach {
```

Comment:
should we add the length check here?

```
assert(expected.length === collectErrors.length)
```

Reply:
we could, but I have found better to have separate test for this, not a big deal though

Reply:
I think this is part of the contract of zip. Because zip bails early on the shorter list it's seems better to check at the source of the problem. Not a biggie.